### ISSUE TITLE
feat: before swap cosmwasm hooks for CL pools

### ISSUE THREAD

**Issue by p0mvn** [osmosis-labs/osmosis#6705]
## Background

Implement hooks for CL pools that call into a cosmwasm contract before every swap.

Should consider if it makes sense to define this at the poolmanager level of abstraction.

This can be implemented by having a parameter of contract addresses controlled by governance. Before every swap, the chain code executes a fire and forget Sudo message to every contract.

The proposal above is up for debate and should be further confirmed with @sunnya97 before starting work on this.

## Acceptance Criteria
- specific design is defined and confirmed 
- before swap CW hooks for CL pools implemented
- thoroughly tested

**Comment by sunnya97** [comment]
Yeah I suppose his actually does make sense at the poolmanager level.

Otoh, we might want to also add hooks for CL for the creating and removing of positions as well, and those will def have to go into the CL module.
